pkg/service/v0/service.go

```diff
@@ -256,13 +330,35 @@ func cleanUpResource(c context.Context, resource *proto.Resource) {
 // the result of this function will always be a valid lower-case UUID or an empty string.
 func getValidatedAccountUUID(c context.Context, accountUUID string) string {
 	if accountUUID == "me" {
```

Comment:
Is the condition correct here, because if accountUUID == "me" then in line 337 it will be overwritten.

Reply:
If the accountUUID is "me", it will be exchanged with the account uuid of the authenticated user. That is part of the context (line 333 extracts it from the context).
Only if that replacement was not successful (because the account uuid of the authenticated user is not in the context) there is a second check in line 337 to clear it then, because the result must be either a) an actually existing account uuid or b) empty. "me" doesn't fulfill this requirement, that's why it's cleared if not successfully extracted from the context.

There are situations where we provide a valid account uuid in the first place, not the string "me". In those cases this function doesn't do anything, just returns the input value.

Reply:
Ah of course, in line 334 the `accountUUID` gets replaced by the one from the context... For some reason I missed that... :+1: